src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -30,24 +31,35 @@ namespace Akka.Persistence.MongoDb.Journal
     public class MongoDbJournal : AsyncWriteJournal
     {
         private static readonly BsonTimestamp ZeroTimestamp = new BsonTimestamp(0);
-        
+
         private readonly MongoDbJournalSettings _settings;
         private Lazy<IMongoDatabase> _mongoDatabase;
         private Lazy<IMongoCollection<JournalEntry>> _journalCollection;
         private Lazy<IMongoCollection<MetadataEntry>> _metadataCollection;
 
-        private ImmutableDictionary<string, IImmutableSet<IActorRef>> _persistenceIdSubscribers = ImmutableDictionary.Create<string, IImmutableSet<IActorRef>>();
-        private ImmutableDictionary<string, IImmutableSet<IActorRef>> _tagSubscribers = ImmutableDictionary.Create<string, IImmutableSet<IActorRef>>();
-        private readonly HashSet<IActorRef> _newEventsSubscriber = new HashSet<IActorRef>();
-        
+        private ImmutableDictionary<string, IImmutableSet<IActorRef>> _persistenceIdSubscribers =
+            ImmutableDictionary.Create<string, IImmutableSet<IActorRef>>();
+
+        private ImmutableDictionary<string, IImmutableSet<IActorRef>> _tagSubscribers =
+            ImmutableDictionary.Create<string, IImmutableSet<IActorRef>>();
+
+        private readonly HashSet<IActorRef> _newEventsSubscriber = new();
+
+        /// <summary>
+        /// Used to cancel all outstanding commands when the actor is stopped.
+        /// </summary>
+        private readonly CancellationTokenSource _pendingCommandsCancellation = new();
```

Comment:
Similar to what we do in Akka.Persistence.Sql.Common - have a global CTS for aborting all in-process operations when the `Journal` shuts down. This same architecture is mirrored in the `SnapshotStore` as well.